src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -669,34 +669,40 @@ private ModuleToken HandleToModuleToken(ref CORINFO_RESOLVED_TOKEN pResolvedToke
             // If the method body is synthetized by the compiler (the definition of the MethodIL is not
             // an EcmaMethodIL), the tokens in the MethodIL are not actual tokens: they're just
             // "per-MethodIL unique cookies". For ready to run, we need to be able to get to an actual
-            // token to refer to the result of token lookup in the R2R fixups.
-            //
-            // We replace the token with the token of the ECMA entity. This only works for **non-generic
```

Comment:
The rest of the paragraph was still valid, it was really just about the `non-generic` word (it wouldn't be legal to encode the token outside the version bubble). But there's an assert that is guarding the version bubble aspect, so maybe the comment is not necessary. I only wrote it because I found the code a bit hard to swallow.

Reply:
Fixed in 3rd commit.